[export][retry] Move lifted tensors out of state_dict

angelayi opened the pull request:
Fixes https://github.com/pytorch/pytorch/issues/110484

Test Plan: CI

Differential Revision: D51321532




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @suo @ydwu4

tugsbayasgalan commented on torch/_export/exported_program.py:
```diff
@@ -216,6 +228,13 @@ def _construct_inp_pos_to_param_buffer_name(new_gm, graph_signature, state_dict)
+        assert hasattr(graph_signature, "lifted_tensor_constants")
+        for name, value in tensor_constants.items():
+            if name in graph_signature.lifted_tensor_constants:
+                new_gm.register_buffer(name, value)
```
Where do we delete the buffers from state_dict?

angelayi committed:
[export][retry] Move lifted tensors out of state_dict (#113689)

Summary:

X-link: https://github.com/pytorch/executorch/pull/1210

Test Plan: CI

Reviewed By: ydwu4

Differential Revision: D51321532
